atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -120,67 +129,87 @@ public synchronized void storeUpperLimit(final long limit) {
         clientPool.runWithRetry(new FunctionCheckedException<Client, Void, RuntimeException>() {
             @Override
             public Void apply(Client client) {
-                cas(client, currentLimit, limit);
+                cas(client, makeNewColumn(currentLimit), currentLimit, limit);
                 return null;
             }
         });
     }
 
-    private void cas(Client client, Long oldVal, long newVal) {
-        final CASResult result;
+    private void cas(Client client, Column oldColumn, Long oldVal, long newVal) {
+        CASResult result = updateTimestampInDb(client, oldColumn, oldVal, newVal);
+        if (!result.isSuccess()) {
+            if (result.getCurrent_values().size() == 0) {
```

Comment:
This method is rather long - we could factor out the failure handling into a new method (possibly multiple methods).

Since I can anticipate the oldColumn, oldVal and newVal being passed around a lot, it might make sense to either use `CheckAndSetRequest`, or to extract a method _object_ whose single responsibility is to make a specific `cas` call and handle the result, so that these passed-around parameters become fields.